core/ustring.cpp

```diff
@@ -2945,11 +2945,11 @@ String String::left(int p_pos) const {
 
 String String::right(int p_pos) const {
 
-	if (p_pos >= size())
-		return *this;
+	if (p_pos >= length())
+		return "";
 
 	if (p_pos < 0)
```

Comment:
I guess it could be `p_pos <= 0` like for `left`?

Reply:
The result is the same but having consistency never hurt anyone. I'm changing it.